SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1239,9 +1240,17 @@ extension PositioningViewController {
         if (!self.isUserNavigating()) {
             self.changeNavigationButtonVisibility(isVisible: true)
         }
+        var secondaryLabel = ""
+        if let buildingInfo = buildingInfo, let poi = poiMarker.poi {
+            if let floor = poi.floor(activeBuildingInfo: buildingInfo) {
+                secondaryLabel = buildingInfo.buildingFloorDescription(floor)
+            } else {
+                secondaryLabel = self.buildingName
+            }
+        }
         self.updateInfoBarLabelsIfNotInsideRoute(
             mainLabel: poiMarker.poi?.name ?? DEFAULT_POI_NAME,
-            secondaryLabel: self.buildingName
+            secondaryLabel: secondaryLabel
```

Comment:
Try to make a method to retrieve the floor an unify with containerInfoBarMap?.setLabels